apps/test-suite/tests/FileSystemNext.ts

```diff
@@ -0,0 +1,95 @@
+'use strict';
+import * as FS from 'expo-file-system';
+import { File, Directory } from 'expo-file-system/next';
+import { Platform } from 'react-native';
+
+export const name = 'FileSystem@next';
+
+export async function test({ describe, expect, it, ...t }) {
+  const testDirectory = FS.documentDirectory + 'tests/';
+  t.beforeEach(async () => {
+    try {
+      await FS.makeDirectoryAsync(testDirectory);
+    } catch {}
+  });
+  t.afterEach(async () => {
+    try {
+      await FS.deleteAsync(testDirectory);
+    } catch {}
+  });
+
+  describe('FileSystem (Next)', () => {
+    it('Creates a lazy file reference', async () => {
+      const file = new File('file:///path/to/file');
+      // The path is normalized by the OS, with both being valid urls (https://stackoverflow.com/a/44725349)
+      if (Platform.OS === 'ios') {
+        expect(file.path).toBe('file:///path/to/file');
+      } else if (Platform.OS === 'android') {
+        expect(file.path).toBe('file:/path/to/file');
+      }
+    });
+
+    it('Allows changing the path property', async () => {
+      const file = new File('file:///path/to/file');
+      file.path = 'file:///new/path';
+      if (Platform.OS === 'ios') {
+        expect(file.path).toBe('file:///new/path');
+      } else if (Platform.OS === 'android') {
+        expect(file.path).toBe('file:/new/path');
+      }
+    });
+
+    it('Writes a string to a file reference', async () => {
+      // Not doing concating path segments in constructor, to make sure the second argument can be an options dict.
+      // Instead, we want to provide utilties for it in a path object.
+      const outputFile = new File(testDirectory + 'file.txt');
```

Comment:
See previous comment

Reply:
I agree, we should discourage from using `+` to concatenate paths. You never know if `testDirectory` ends with `/` or not.